drivers/netdev2_ieee802154/netdev2_ieee802154.c

```diff
@@ -129,19 +106,13 @@ int netdev2_ieee802154_get(netdev2_ieee802154_t *dev, netopt_t opt, void *value,
             break;
 #ifdef MODULE_GNRC
         case NETOPT_PROTO:
-            if (max_len < sizeof(gnrc_nettype_t)) {
-                res = -EOVERFLOW;
-                break;
-            }
+            assert(max_len == sizeof(gnrc_nettype_t));
             *((gnrc_nettype_t *)value) = dev->proto;
             res = sizeof(gnrc_nettype_t);
             break;
 #endif
         case NETOPT_DEVICE_TYPE:
-            if (max_len < sizeof(uint16_t)) {
-                res = -EOVERFLOW;
-                break;
-            }
+            assert(max_len == sizeof(uint16_t));
```

Comment:
To say it in another way: we have two options here: assume a type (which per doc we do) or guess the type and act upon it (which is more complicated and would result in the opposite we are trying to do by changing ifs to asserts (reducing code size).